lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -1391,13 +1389,54 @@ class BuildControl : public calyx::FuncOpPartialLoweringPattern {
         rewriter.setInsertionPointToStart(callBody.getBodyBlock());
         std::string initGroupName = "init_" + instanceOp.getSymName().str();
         rewriter.create<calyx::EnableOp>(instanceOp.getLoc(), initGroupName);
+
+        auto callee = callSchedPtr->callOp.getCallee();
+        auto *calleeOp = SymbolTable::lookupNearestSymbolFrom(
+            callSchedPtr->callOp.getOperation()->getParentOp(),
+            StringAttr::get(rewriter.getContext(), "func_" + callee.str()));
+        FuncOp calleeFunc = dyn_cast_or_null<FuncOp>(calleeOp);
+
+        auto instanceOpComp =
+            llvm::cast<calyx::ComponentOp>(instanceOp.getReferencedComponent());
+        auto *instanceOpLoweringState =
+            loweringState().getState(instanceOpComp);
+
         SmallVector<Value, 4> instancePorts;
-        auto inputPorts = callSchedPtr->callOp.getOperands();
+        SmallVector<Value, 4> inputPorts;
+        SmallVector<Attribute, 4> refCells;
+        for (auto operandEnum : enumerate(callSchedPtr->callOp.getOperands())) {
+          auto operand = operandEnum.value();
+          auto index = operandEnum.index();
+          if (isa<MemRefType>(operand.getType())) {
+            auto memOpName = getState<ComponentLoweringState>()
+                                 .getMemoryInterface(operand)
+                                 .memName();
+            auto memOpNameAttr =
+                SymbolRefAttr::get(rewriter.getContext(), memOpName);
+            Value argI = calleeFunc.getArgument(index);
+            if (isa<MemRefType>(argI.getType())) {
+              NamedAttrList namedAttrList;
+              namedAttrList.append(
+                  rewriter.getStringAttr(
+                      instanceOpLoweringState->getMemoryInterface(argI)
+                          .memName()),
+                  memOpNameAttr);
+              refCells.push_back(
+                  DictionaryAttr::get(rewriter.getContext(), namedAttrList));
+            }
+          } else {
+            inputPorts.push_back(operand);
+          }
+        }
```

Comment:
```suggestion
        for (auto operandEnum : enumerate(callSchedPtr->callOp.getOperands())) {
          auto operand = operandEnum.value();
          auto index = operandEnum.index();
          if (!isa<MemRefType>(operand.getType())) {
            inputPorts.push_back(operand);
            continue;
          }
          auto memOpName = getState<ComponentLoweringState>()
                                 .getMemoryInterface(operand)
                                 .memName();
          auto memOpNameAttr =
                SymbolRefAttr::get(rewriter.getContext(), memOpName);
          Value argI = calleeFunc.getArgument(index);
          if (isa<MemRefType>(argI.getType())) {
            NamedAttrList namedAttrList;
            namedAttrList.append(
                rewriter.getStringAttr(
                    instanceOpLoweringState->getMemoryInterface(argI)
                        .memName()),
                memOpNameAttr);
            refCells.push_back(
                DictionaryAttr::get(rewriter.getContext(), namedAttrList));
          }
        }
```

Reply:
We should just use an early exist in the case where this isn't a MemRefType. 

Ref: https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code